### PR TITLE
 Stress test with send_columns for Rust & C++, improve C++ TimeColumn api, fix reporting

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* re-opened https://github.com/rerun-io/rerun/pull/7148
    * sth strange going on with Github right now.
* Add `send_column` variant to plot stress tests for C++ and Rust.
* Small addition to C++ `TimeColumn`: `from_seconds`
* `from_times_nanoseconds` -> `from_nanoseconds`
* fix bugs in reporting on all languages

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7148?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7148?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7148)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Review by teh-cmc — CHANGES_REQUESTED** [review]
Look for red squares :red_square: to see what's wrong.

## Rust

### Standard

```
pixi run rs-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000
```
```
logged 49950 scalars over 1.000558926s (freq=49922.097Hz, expected=50000.000Hz, load=319.419%)
logged 50000 scalars over 1.000603315s (freq=49969.852Hz, expected=50000.000Hz, load=80.576%)
logged 50000 scalars over 1.000785875s (freq=49960.737Hz, expected=50000.000Hz, load=77.516%)
logged 50000 scalars over 1.000822225s (freq=49958.923Hz, expected=50000.000Hz, load=62.591%)
logged 50000 scalars over 1.001024724s (freq=49948.816Hz, expected=50000.000Hz, load=94.965%)
logged 50 scalars over 2.807928ms (freq=17806.724Hz, expected=50000.000Hz, load=28.245%)
```

- [x] reports 50kHz in the logs
- [x] reports 250k points in the logs
- [x] 250k points in memory panel

:white_check_mark: all good

### Batch

```
pixi run rs-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000 --temporal-batch-size 1000
```
```
logged 100000 scalars over 1.001562323s (freq=99844.011Hz, expected=50000.000Hz, load=0.143%)
logged 50000 scalars over 1.001339216s (freq=49933.129Hz, expected=50000.000Hz, load=0.158%)
logged 50000 scalars over 1.001358009s (freq=49932.192Hz, expected=50000.000Hz, load=0.137%)
logged 50000 scalars over 1.001560832s (freq=49922.080Hz, expected=50000.000Hz, load=0.140%)
logged 0 scalars over 1.000246371s (freq=0.000Hz, expected=50000.000Hz, load=0.162%)
```

- [ ] reports 50kHz in the logs
- [x] reports 250k points in the logs
- [x] 250k points in memory panel

:red_square: the first tick runs at 100kHz

## C++

### Standard

```
pixi run cpp-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000
```
```
logged 50000 scalars over 1.00063s (freq=49968.6Hz, expected=50000Hz, load=237.549%)
logged 50000 scalars over 1.00016s (freq=49992Hz, expected=50000Hz, load=107.678%)
logged 50000 scalars over 1.00002s (freq=49999.2Hz, expected=50000Hz, load=91.5501%)
logged 50000 scalars over 1.00012s (freq=49994Hz, expected=50000Hz, load=93.6422%)
logged 50000 scalars over 1.00051s (freq=49974.7Hz, expected=50000Hz, load=117.017%)
logged 0 scalars over 0.000474865s (freq=0Hz, expected=50000Hz, load=58.1364%)
```

- [x] reports 50kHz in the logs
- [x] reports 250k points in the logs
- [x] 250k points in memory panel

:white_check_mark: all good

### Batch

```
pixi run cpp-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000 --temporal-batch-size 1000
```
```
logged 100000 scalars over 1.00173s (freq=99827.3Hz, expected=50000Hz, load=0.152773%)
logged 100000 scalars over 1.99962s (freq=50009.6Hz, expected=50000Hz, load=0.178009%)
logged 50000 scalars over 1.00004s (freq=49998Hz, expected=50000Hz, load=0.142315%)
logged 0 scalars over 0.998852s (freq=0Hz, expected=50000Hz, load=0.148753%)
```

- [ ] reports 50kHz in the logs
- [x] reports 250k points in the logs
- [x] 250k points in memory panel

:red_square: the first two ticks run at 100kHz

## Python

### Standard

```
pixi run py-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000
```
```
logged 27650 scalars over 1.002s (freq=27606.967Hz, expected=50000.0Hz, load=847.507%)
logged 25950 scalars over 1.001s (freq=25928.858Hz, expected=50000.0Hz, load=299.931%)
logged 23200 scalars over 1.002s (freq=23151.492Hz, expected=50000.0Hz, load=388.789%)
logged 22550 scalars over 1.001s (freq=22536.777Hz, expected=50000.0Hz, load=404.954%)
logged 22950 scalars over 1.001s (freq=22932.324Hz, expected=50000.0Hz, load=391.722%)
logged 24650 scalars over 1.002s (freq=24611.39Hz, expected=50000.0Hz, load=336.766%)
logged 23550 scalars over 1.0s (freq=23549.433Hz, expected=50000.0Hz, load=398.159%)
logged 24300 scalars over 1.002s (freq=24253.618Hz, expected=50000.0Hz, load=394.559%)
logged 23950 scalars over 1.001s (freq=23932.215Hz, expected=50000.0Hz, load=400.853%)
logged 24100 scalars over 1.0s (freq=24091.436Hz, expected=50000.0Hz, load=397.706%)
logged 7150 scalars over 0.294s (freq=24297.115Hz, expected=50000.0Hz, load=301.957%)
```

- [x] reports 50kHz in the logs
- [x] reports 250k points in the logs
- [x] 250k points in memory panel

:white_check_mark: all good -- doesn't hit 50kHz but that's expected

### Batch

```
pixi run py-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000 --temporal-batch-size 1000
```
```
Traceback (most recent call last):
  File "/home/cmc/dev/rerun-io/rerun/tests/python/plot_dashboard_stress/main.py", line 237, in <module>
    main()
  File "/home/cmc/dev/rerun-io/rerun/tests/python/plot_dashboard_stress/main.py", line 175, in main
    time_column = rr.TimeBatch("sim_time", sim_time)
                  ^^^^^^^^^^^^
AttributeError: module 'rerun' has no attribute 'TimeBatch'
```

:red_square: doesn't build

File: rerun_cpp/src/rerun/time_column.hpp [hunk 1]
```diff
@@ -72,25 +72,31 @@ namespace rerun {
             );
         }
 
-        /// Creates a sequence time column from an array of sequence points.
+        /// Creates a sequence time column from an array of nanoseconds.
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply rerun_cpp/src/rerun/time_column.hpp:75]: your suggestion doesn't contain anything, but I see the problem 😄

**Comment by Wumpf** [comment]
addressing the build issue on python makes it show the "🟥 the first `N` ticks run at 100kHz" issue as well.
Digging!

**Comment by Wumpf** [comment]
ran the full matrix again, all cases work now correctly

File: tests/python/plot_dashboard_stress/main.py [hunk 2]
```diff
@@ -183,31 +183,19 @@ def main() -> None:
                 else:
                     value_index = slice(index, index + args.temporal_batch_size)
                     value_batch = rr.components.ScalarBatch(values[value_index, plot_idx, series_idx])
-                    rr.send_columns(
+                    rr.send_columnsstorage_type(
```
Comment (teh-cmc) [inline tests/python/plot_dashboard_stress/main.py:186]:
```suggestion
                    rr.send_columns(

**Review by teh-cmc — APPROVED** [review]
very nice

File: tests/python/plot_dashboard_stress/main.py (hunk 2 above)
Comment (Wumpf) [reply tests/python/plot_dashboard_stress/main.py]: how did I...

**Pull request merged**